client/asset/eth/eth.go

```diff
@@ -3733,42 +3733,42 @@ func (eth *ETHWallet) checkForNewBlocks(ctx context.Context) {
 	}
 }
 
-// ConfirmRedemption checks the status of a redemption. If a transaction has
-// been fee-replaced, the caller is notified of this by having a different
-// coinID in the returned asset.ConfirmRedemptionStatus as was used to call the
+// ConfirmTransaction checks the status of a redemption or refund. If a tx
+// has been fee-replaced, the caller is notified of this by having a different
+// coinID in the returned asset.ConfirmTxStatus as was used to call the
 // function. Fee argument is ignored since it is calculated from the best
 // header.
-func (w *ETHWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption, _ uint64) (*asset.ConfirmRedemptionStatus, error) {
-	return w.confirmRedemption(coinID, redemption)
+func (w *ETHWallet) ConfirmTransaction(coinID dex.Bytes, confirmTx *asset.ConfirmTx, _ uint64) (*asset.ConfirmTxStatus, error) {
+	return w.confirmTransaction(coinID, confirmTx)
 }
 
-// ConfirmRedemption checks the status of a redemption. If a transaction has
-// been fee-replaced, the caller is notified of this by having a different
-// coinID in the returned asset.ConfirmRedemptionStatus as was used to call the
+// ConfirmTransaction checks the status of a redemption or refund. If a tx
+// has been fee-replaced, the caller is notified of this by having a different
+// coinID in the returned asset.ConfirmTxStatus as was used to call the
 // function. Fee argument is ignored since it is calculated from the best
 // header.
-func (w *TokenWallet) ConfirmRedemption(coinID dex.Bytes, redemption *asset.Redemption, _ uint64) (*asset.ConfirmRedemptionStatus, error) {
-	return w.confirmRedemption(coinID, redemption)
+func (w *TokenWallet) ConfirmTransaction(coinID dex.Bytes, confirmTx *asset.ConfirmTx, _ uint64) (*asset.ConfirmTxStatus, error) {
+	return w.confirmTransaction(coinID, confirmTx)
 }
 
-func confStatus(confs, req uint64, txHash common.Hash) *asset.ConfirmRedemptionStatus {
-	return &asset.ConfirmRedemptionStatus{
+func confStatus(confs, req uint64, txHash common.Hash) *asset.ConfirmTxStatus {
+	return &asset.ConfirmTxStatus{
 		Confs:  confs,
 		Req:    req,
 		CoinID: txHash[:],
 	}
 }
 
-// confirmRedemption checks the confirmation status of a redemption transaction.
-func (w *assetWallet) confirmRedemption(coinID dex.Bytes, redemption *asset.Redemption) (*asset.ConfirmRedemptionStatus, error) {
+// confirmTransaction checks the confirmation status of a transaction.
+func (w *assetWallet) confirmTransaction(coinID dex.Bytes, confirmTx *asset.ConfirmTx) (*asset.ConfirmTxStatus, error) {
 	if len(coinID) != common.HashLength {
 		return nil, fmt.Errorf("expected coin ID to be a transaction hash, but it has a length of %d",
 			len(coinID))
 	}
 	var txHash common.Hash
 	copy(txHash[:], coinID)
 
```

Comment:
Refund may insert a zero hash if already refunded. We should confirm that here since we cannot check it.

Reply:
done